examples/website-demos/social-messaging/src/components/MessageInput/SocialMessageInput.tsx

```diff
@@ -144,13 +163,7 @@ export const SocialMessageInput = (props: Props) => {
           className={`input-ui-send ${text || numberOfUploads ? 'text' : ''}`}
           onClick={handleSubmit}
         >
-          {giphyState && !numberOfUploads ? (
-            <GiphySearch />
-          ) : (
-            <>
-              <SendArrow />
-            </>
-          )}
+          {giphyState && !numberOfUploads ? <GiphySearch /> : <>{renderSendButton()}</>}
```

Comment:
instead of invoking the function in the jsx we can do the conditional outside the return so we create a constant reference to the component, so something like

```
const SendButton = cooldownRemaining ? (
      <SocialCooldownTimer
        cooldownInterval={cooldownInterval}
        setCooldownRemaining={setCooldownRemaining}
      />
    ) : (
      <SendArrow />
    );
  };
```

Reply:
and i guess you could even toss it in a useMemo with cooldownRemaining as the dependency so it doesn't recompute every other stuff in the component changes, like text for example

Reply:
good call - updated.